src/Component.php

```diff
@@ -5,70 +5,83 @@
+
+    protected function run(): void
+    {
         $dataFolder = $this->getDataDir();
 
         /** @var Config $config */
         $config = $this->getConfig();
         $configRawData = $config->getRawConfig();
-        $outputPath = $dataFolder . "/out/tables";
+        $outputPath = $dataFolder . '/out/tables';
 
         $parameters = $this->getConfig()->getParameters();
 
         $finder = new Finder();
-        $finder->notName("*.manifest")->in($dataFolder . "/in/tables")->depth(0);
+        $finder->notName('*.manifest')->in($dataFolder . '/in/tables')->depth(0);
 
         foreach ($finder as $sourceFile) {
             $manifest = $this->getManifestManager()->getTableManifest($sourceFile->getBasename());
 
-            $manifestVariables = array_keys($manifest);
             $configVariables = [];
-            if (isset($configRawData["parameters"]) && is_array($configRawData["parameters"])) {
-                $configVariables = array_keys($configRawData["parameters"]);
+            if (isset($configRawData['parameters'])) {
+                $configVariables = array_keys($configRawData['parameters']);
             }
 
             /*
              * If the manifest value is not set or the user provided value is set (even empty), set the manifest
              * to the parameters value (= user provided value or default)
              * https://github.com/keboola/processor-create-manifest/pull/11#discussion_r185779866
              */
-            if (!in_array("delimiter", $manifestVariables) || in_array("delimiter", $configVariables)) {
-                $manifest["delimiter"] = $parameters["delimiter"];
+            if ($manifest->getDelimiter() === null || in_array('delimiter', $configVariables)) {
+                $manifest->setDelimiter($parameters['delimiter'] ?? self::DEFAULT_DELIMITER);
             }
-            if (!in_array("enclosure", $manifestVariables) || in_array("enclosure", $configVariables)) {
-                $manifest["enclosure"] = $parameters["enclosure"];
+            if ($manifest->getEnclosure() === null || in_array('enclosure', $configVariables)) {
+                $manifest->setEnclosure($parameters['enclosure'] ?? self::DEFAULT_ENCLOSURE);
             }
-            if (!in_array("primary_key", $manifestVariables) || in_array("primary_key", $configVariables)) {
-                $manifest["primary_key"] = $parameters["primary_key"];
+
+            if ($manifest->getSchema() === [] || in_array('columns', $configVariables)) {
+                $manifest->setSchema([]);
+                foreach ($parameters['columns'] as $column) {
+                    $manifest->addSchema(new ManifestOptionsSchema($column));
+                }
             }
-            if (!in_array("incremental", $manifestVariables) || in_array("incremental", $configVariables)) {
-                $manifest["incremental"] = $parameters["incremental"];
+
+            if (in_array('primary_key', $configVariables)) {
```

Comment:
Tady by se imho mělo použít to LegacyPrimaryKeys pokud nejsou sloupce vyjmenovaný. Jinak by nešel vygenerovat nový typ manifestu a fallbacknulo by to na starý. Ideálně to přidej do testů, přijde mi, že tam primaryKeys nikde nejsou pokrytý

Reply:
Jak přesně to myslíš? setLegacyPrimaryKeys se tu volá
Nebo by se mělo používat setPrimaryKeyColumns a jen pokud sloupce nejsou tak to setLegacyPrimaryKeys?
Primární klíče jsou pokryté testy primary-key-old-output-manifest a primary-key-authoritative-data-type-forced-to-old-manifest

Reply:
`Nebo by se mělo používat setPrimaryKeyColumns a jen pokud sloupce nejsou tak to setLegacyPrimaryKeys` jj přesně tak

Reply:
jasný, upravím